### PR TITLE
Issue 3705 site logo ur is

### PR DESCRIPTION
**Pull request by SamPenrose**
1) Supports https: URIs from any domain while excluding http: and relative URIs.
2) Supports data:image/ URIs.
3) Does NOT restrict the latter by size.
4) Does NOT support app: URIs.
5) RP parameter validation factored into modules/validate_rp_params by @shane-tomlinson .

### CODE REVIEW

File: resources/static/dialog/js/modules/validate_rp_params.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    // into an array that looks like:
+    //   ['data:image/png;base64,iV...', 'image', 'png', ...]
+    // which means that mimetype proper is represented as-> [1]/[2]
+    var dataUriRegex = /^data:(.+)\/(.+);base64,(.*)$/;
+
+    dataMatches = inputLogoUri.match(dataUriRegex);
+    if (dataMatches) {
+      if ((dataMatches[1].toLowerCase() === 'image')
+           &&
+          (imageMimeTypes.indexOf(dataMatches[2].toLowerCase()) > -1)) {
```
Comment (jaredhirsch) [inline resources/static/dialog/js/modules/validate_rp_params.js]: use _.indexOf for ie8 compat

**Comment by SamPenrose** [comment]
indexOf fix pushed

**Comment by jaredhirsch** [comment]
@SamPenrose thanks for the quick IE8 fix! I think this looks good, but the refactor gives it serious size. I want to get it on dev and poke at it, seems like a lot of changes to merge in just before branching.

I'll branch the train tonight, re-read this once more in the morning, and merge after that, modulo any final nits.

**Comment by shane-tomlinson** [comment]
@SamPenrose - I'm a bit late to the party, have you checked large (>32KB) data URIs on IE8?

**Comment by SamPenrose** [comment]
Nope. Perhaps foolishly took silence for consent on not restricting size.

**Pull request merged**